extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/internal/XMLResponseHandler.java

```diff
@@ -369,7 +375,7 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
                 break;
             case Preset:
                 if (state == XMLHandlerState.Presets) {
-                    commandExecutor.addContentItemToPresetContainer(contentItem.getPresetID(), contentItem);
+                    playerPresets[contentItem.getPresetID() - 1] = contentItem;

```

Comment:
is `getPresetID` guaranteed to return something in the range of `[1,6]`? Or would it be more safe to switch `playerPresets` to a map directly? This would also make the `updatePresetContainerFromPlayer` method more secure when dealing with a map?
In `updatePresetContainerFromPlayer` you essentially replace the options known by the `presetContainer` in the handler with the options you process here.

Reply:
That would actually be tricky because the player returns only the items currently stored. This means if there is an item stored on position 3 - only that position will be returned and we should manually handle clearing positions 1, 2, 4, 5 and 6.

Reply:
Thats what I meant with 

> replace the options

When only the option at position 3 is returned and it is the only option which should presented to the user why not simply replace all options known in the `presetContainer`? None of them should be retained, so you can simply overwrite with the result from the speaker.

Reply:
Because the presetContainer can also contain values above number 6 (which are not stored on the player but are still possible to play) and I would like to preserve them.

Reply:
Thanks, that was not clear to me.